.circleci/config.yml

```diff
@@ -0,0 +1,103 @@
+version: 2
+
+defaults: &defaults
+  working_directory: ~/project/yarn
+  docker:
+    - image: yarnpkg/dev:latest
+
+jobs:
+  install:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          key: yarn-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - run:
+          name: Install Dependencies
+          command: yarn install
+      - save_cache:
+          key: yarn-{{ .Branch }}-{{ checksum "yarn.lock" }}
+          paths:
+            - .cache/yarn
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - yarn
+  test:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Tests
+          command: |
+            node -v
+            if [ "$CIRCLE_BRANCH" == 'master' ]; then
+              ./scripts/set-dev-version.js
+            fi;
+            yarn lint
+            # Limit maxWorkers to 3 to avoid OOM on CircleCI
+            yarn test-ci -- -- --maxWorkers 3
+            yarn check-lockfile
+  build:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Build distribution
+          command: |
+            node -v
+            if [ "$CIRCLE_BRANCH" == 'master' ]; then
+              ./scripts/set-dev-version.js
+            fi;
+            yarn build-dist
+            ./scripts/build-deb.sh
+      - store_artifacts:
+          path: artifacts/
+          destination: yarnpkg
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - yarn
+  deploy:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Deploy
+          command: |
+            # Only NPM is handled here - All other release files are handled in a webhook.
+            if [ "${CIRCLE_PROJECT_USERNAME}" == "yarnpkg" ]; then
```

Comment:
Can you confirm that this is **not** sent for pull requests, and only for builds in the repo itself?

Reply:
Does that mean that we do not want to publish when some feature branches are merged into master?
I want to know the release flow when publish to npm.

on master branch:

```
$ git tag - a v 0.27.0 (example)
$ git push origin master
```

and only waiting for CircleCI?

Reply:
> I want to know the release flow when publish to npm.

Basically what you mentioned. `scripts/release-branch.sh` is what is used for new major or minor releases (eg. `0.29.0` or `1.0.0`)

We want to publish when a tag is created and the repo is the `yarnpkg/yarn` repo (that is, we need to be very careful to **not** publish a pull request from someone else's repo, for example if they tag it and then send a PR). This is likely fine with your changes though. Just trying to be cautious :)

Reply:
Thank you for more information.

> for example if they tag it and then send a PR

We can't include the tag we've created into the Pull Request. Therefore, there is no worry that external people tagging.
https://stackoverflow.com/questions/12278660/adding-tags-to-a-pull-request